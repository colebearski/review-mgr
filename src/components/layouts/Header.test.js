import React from "react";
// Shallow: render top level component
// Deep: renders all of the nested components
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

import Header from "./Header";

describe("Header", () => {
  test("Rendering the functional header component", () => {
    const wrapper = shallow(<Header />);

    expect(wrapper).toMatchSnapshot();
  });
});
