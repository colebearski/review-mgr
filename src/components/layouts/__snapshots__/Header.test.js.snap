// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Rendering the functional header component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    branding="Shakespeare Reviews"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          Shakespeare Reviews
        </a>
        <div>
          <ul
            className="navbar-nav mr-auto"
          >
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/"
              >
                Home
              </a>
            </li>
          </ul>
        </div>
      </div>,
      "className": "navbar navbar-expand-sm navbar-dark bg-info mb-3 py-0",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="navbar-brand"
            href="/"
          >
            Shakespeare Reviews
          </a>,
          <div>
            <ul
              className="navbar-nav mr-auto"
            >
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/"
                >
                  Home
                </a>
              </li>
            </ul>
          </div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Shakespeare Reviews",
            "className": "navbar-brand",
            "href": "/",
          },
          "ref": null,
          "rendered": "Shakespeare Reviews",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="navbar-nav mr-auto"
            >
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/"
                >
                  Home
                </a>
              </li>
            </ul>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/"
                >
                  Home
                </a>
              </li>,
              "className": "navbar-nav mr-auto",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="nav-link"
                  href="/"
                >
                  Home
                </a>,
                "className": "nav-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Home",
                  "className": "nav-link",
                  "href": "/",
                },
                "ref": null,
                "rendered": "Home",
                "type": "a",
              },
              "type": "li",
            },
            "type": "ul",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <a
            className="navbar-brand"
            href="/"
          >
            Shakespeare Reviews
          </a>
          <div>
            <ul
              className="navbar-nav mr-auto"
            >
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/"
                >
                  Home
                </a>
              </li>
            </ul>
          </div>
        </div>,
        "className": "navbar navbar-expand-sm navbar-dark bg-info mb-3 py-0",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="navbar-brand"
              href="/"
            >
              Shakespeare Reviews
            </a>,
            <div>
              <ul
                className="navbar-nav mr-auto"
              >
                <li
                  className="nav-item"
                >
                  <a
                    className="nav-link"
                    href="/"
                  >
                    Home
                  </a>
                </li>
              </ul>
            </div>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Shakespeare Reviews",
              "className": "navbar-brand",
              "href": "/",
            },
            "ref": null,
            "rendered": "Shakespeare Reviews",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ul
                className="navbar-nav mr-auto"
              >
                <li
                  className="nav-item"
                >
                  <a
                    className="nav-link"
                    href="/"
                  >
                    Home
                  </a>
                </li>
              </ul>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <li
                  className="nav-item"
                >
                  <a
                    className="nav-link"
                    href="/"
                  >
                    Home
                  </a>
                </li>,
                "className": "navbar-nav mr-auto",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    className="nav-link"
                    href="/"
                  >
                    Home
                  </a>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Home",
                    "className": "nav-link",
                    "href": "/",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": "a",
                },
                "type": "li",
              },
              "type": "ul",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
